src/Equations/SPIHandler.cpp

```diff
@@ -327,11 +327,11 @@ void SPIHandler::Rebuild(real_t dt){
         // Shift the deposition profile
         for(len_t ip=0;ip<nShard;ip++){
             if(rCoordPNext[ip]>rCoordPPrevious[ip]){
-                for(len_t ir=0;ir<nr-1;ir++){
+                for(len_t ir=0;ir<nr-nbrShiftGridCell[ip];ir++){
```

Comment:
I suppose this is a borrow from the ``plasmoid_drift`` branch? But is it correct? Shouldn't ``ir`` be affected more directly? And do we really want these changes on this branch, before the ``plasmoid_drift`` branch is merged?

Reply:
Discuss!